go.mod

```diff
@@ -217,3 +217,219 @@ require (
 	sigs.k8s.io/yaml v1.1.0
 	vbom.ml/util v0.0.0-20160121211510-db5cfe13f5cc
 )
+
+replace (
+	bitbucket.org/bertimus9/systemstat => bitbucket.org/bertimus9/systemstat v0.0.0-20180207000608-0eeff89b0690
```

Comment:
I don't get it - these seem to be a duplication of the require block above - what's this about?

Reply:
`require` notes a dependency of k8s.io/kubernetes on the given module at the given version (but go is free to pick a newer version if there are conflicting dependencies anywhere in our dependency graph)

`replace` pins the given module to the given version.

you need both in the top-level module being built to ensure that no dependency silently changes the versions you're pulling in.

Reply:
that seems super pointless - why would we `require` a specific tag only to have go change that under us?  I can see how it might matter for version ranges, but it seems pointless for examct specs?

Reply:
welcome to go module [minimum version selection](https://github.com/golang/go/wiki/Modules#version-selection):

> The minimal version selection algorithm is used to select the versions of all modules used in a build. For each module in a build, the version selected by minimal version selection is always the semantically highest of the versions explicitly listed by a require directive in the main module or one of its dependencies.